distribution/pull_v2.go

```diff
@@ -413,26 +417,26 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		oldTagID, err := p.config.ReferenceStore.Get(ref)
 		if err == nil {
 			if oldTagID == id {
-				return false, addDigestReference(p.config.ReferenceStore, ref, manifestDigest, id)
+				return imgID, false, addDigestReference(p.config.ReferenceStore, ref, manifestDigest, id)
 			}
 		} else if err != refstore.ErrDoesNotExist {
-			return false, err
+			return imgID, false, err
 		}
 
 		if canonical, ok := ref.(reference.Canonical); ok {
 			if err = p.config.ReferenceStore.AddDigest(canonical, id, true); err != nil {
-				return false, err
+				return imgID, false, err
 			}
 		} else {
 			if err = addDigestReference(p.config.ReferenceStore, ref, manifestDigest, id); err != nil {
-				return false, err
+				return imgID, false, err
 			}
 			if err = p.config.ReferenceStore.AddTag(ref, id, true); err != nil {
-				return false, err
+				return imgID, false, err
 			}
 		}
 	}
-	return true, nil
+	return reference.FamiliarString(id), true, nil
```

Comment:
This looks to be converting the image _id_ into a "digest" format, not the digest of the manifest (see later comment with example)